sources/Launcher/DYNMarginCalculationLauncher.cpp

```diff
@@ -523,24 +531,50 @@ void MarginCalculationLauncher::findOrLaunchScenarios(const std::string& baseJob
   }
 
   for (unsigned int i=0; i < events2Run.size(); i++) {
-    double variation = events2Run[i].second;
+    double variation = events2Run[i].variation_;
     std::string iidmFile = generateIDMFileNameForVariation(variation);
     inputsByIIDM_.erase(iidmFile);  // remove iidm file used for scenario to save RAM
   }
 }
 
+MarginCalculationLauncher::EventPair::EventPair(size_t index, double variation) : index_(index), variation_(variation) {
+}
+
+void MarginCalculationLauncher::addAvailableEvents(const double variation,
+                                                   const std::vector<size_t>& eventsIdNoFilter,
+                                                   const std::vector<double >& maximumVariationPassing,
+                                                   std::vector<EventPair >& events2Run) {
+  if ((*loadIncreaseStatus_.find(variation)).second.success) {
+    auto found = scenarioStatus_.find(variation);
+    unsigned int eventIndex = 0;
+    while (events2Run.size() % mpi::context().nbProcs() != 0  && eventIndex < eventsIdNoFilter.size() &&
+           (variation > maximumVariationPassing[eventsIdNoFilter[eventIndex]] ||
```

Comment:
Why do we take only variation above the maximum?

Reply:
Maybe I am wrong but no need to launch scenarios on variations below a variation for which we already know the scenario is working